crates/uv-auth/src/store.rs

```diff
@@ -75,41 +20,64 @@ impl Default for AuthenticationStore {
 impl AuthenticationStore {
     pub fn new() -> Self {
         Self {
-            credentials: Mutex::new(HashMap::new()),
+            store: Mutex::new(HashMap::new()),
         }
     }
 
-    pub fn get(&self, url: &Url) -> Option<Option<Credential>> {
+    /// Retrieve stored credentials for a URL, if any.
+    pub fn get(&self, url: &Url) -> Option<Arc<Credentials>> {
         let netloc = NetLoc::from(url);
-        let credentials = self.credentials.lock().unwrap();
-        credentials.get(&netloc).cloned()
+        let store = self.store.lock().unwrap();
+        if let Some(fetched) = store.get(&netloc) {
```

Comment:
Can you use `let fetched = store.get(&netloc)?` here instead of manual `if`-`else`?

Reply:
Yeah TIL you can `?` on `Option` lmao

Reply:
Welcome to the `Option` **monad**